src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/Messager.java

```diff
@@ -166,190 +183,189 @@ public Messager(Context context, String programName, PrintWriter outWriter, Prin
-     * Part of DocErrorReporter.
+     * Prints a warning message for a given documentation tree node.
      *
-     * @param msg message to print
+     * @param path    the path for the documentation tree node
+     * @param message the message
      */
-    public void printWarning(String msg) {
-        printWarning((DocTreePath)null, msg);
+    public void printWarning(DocTreePath path, String message) {
+        DiagnosticSource ds = getDiagnosticSource(path);
+        DiagnosticPosition dp = getDiagnosticPosition(path);
+        report(DiagnosticType.WARNING, EnumSet.noneOf(DiagnosticFlag.class), ds, dp, message);
     }
 
-    public void printWarningUsingKey(String key, Object... args) {
-        printWarning((Element)null, getText(key, args));
+    /**
+     * Prints a warning message for a given element.
+     *
+     * @param element the element
+     * @param message the message
+     */
+    public void printWarning(Element element, String message) {
+        DiagnosticSource ds = getDiagnosticSource(element);
+        DiagnosticPosition dp = getDiagnosticPosition(element);
+        report(DiagnosticType.WARNING, EnumSet.noneOf(DiagnosticFlag.class), ds, dp, message);
     }
 
-    public void printWarning(Element e, String key, Object... args) {
+    /**
+     * Prints a warning message.
+     *
+     * @param key the resource key for the message
+     * @param args the arguments for the message
+     */
+    public void printWarningUsingKey(String key, Object... args) {
         printWarning(getText(key, args));
     }
 
-    public void printWarning(DocTreePath path, String msg) {
-        String prefix = getDiagSource(path);
-        if (diagListener != null) {
-            report(DiagnosticType.WARNING, prefix, msg);
-            return;
-        }
-        printWarning(prefix, msg);
-    }
-
-    public void printWarning(Element e, String msg) {
-        String prefix = getDiagSource(e);
-        if (diagListener != null) {
-            report(DiagnosticType.WARNING, prefix, msg);
-            return;
-        }
-        printWarning(prefix, msg);
-    }
-
-    // print the warning and increment count
-    private void printWarning(String prefix, String msg) {
-        if (nwarnings < MaxWarnings) {
-            PrintWriter warnWriter = getWriter(WriterKind.WARNING);
-            printRawLines(warnWriter, prefix + ": " + getText("javadoc.warning") + " - " + msg);
-            warnWriter.flush();
-            nwarnings++;
-        }
+    /**
+     * Prints a warning message for an element.
+     *
+     * @param element the element
+     * @param key     the resource key for the message
+     * @param args    the arguments for the message
+     */
+    public void printWarningUsingKey(Element element, String key, Object... args) {
+        printWarning(element, getText(key, args));
     }
 
     /**
-     * Print a message.
-     * Part of DocErrorReporter.
+     * Print a "notice" message.
```

Comment:
Add "s" to "print" for consistency with doc comment of other methods in this class.

Reply:
Definitely.